wetterdienst/provider/dwd/road_weather/api.py

```diff
@@ -0,0 +1,183 @@
+    create_meta_index_for_road_weather,
+)
+from wetterdienst.provider.dwd.road_weather.parser import parse_dwd_road_weather_data
+from wetterdienst.util.geo import Coordinates, derive_nearest_neighbours
+
+log = logging.getLogger(__name__)
+
+
+class DwdRoadWeatherObservationValues:
+    """
+    The DwdRoadWeatherObservationValues class represents a request for
+    observation data from road weather stations as provided by the DWD service.
+    """
+
+    def __init__(
+        self,
+        period: DwdRoadWeatherObservationPeriod = DwdRoadWeatherObservationPeriod.LATEST,
+    ):
+        self._tz = Timezone.GERMANY
+        self._data_tz = Timezone.UTC
+        self.period = period
+
+        self.metaindex = create_meta_index_for_road_weather()
+
+        # actually new stations with group id XX are not available
+        self.metaindex = self.metaindex[
+            self.metaindex[Columns.STATION_GROUP.value] != "XX"
+        ]
+
+    def __eq__(self, other):
+        """ Add resolution and periods """
+        return super(DwdRoadWeatherObservationValues, self).__eq__(other)
+
+    def _collect_data_by_rank(
+        self,
+        latitude: float,
+        longitude: float,
+        rank: int = 1,
+    ) -> pd.DataFrame:
+        """
+        Takes coordinates and derives the closest stations to the given location.
+        These locations define the required Station Groups to download and parse the data
+        """
+        distances, indices_nearest_neighbours = derive_nearest_neighbours(
+            self.metaindex[Columns.LATITUDE.value].values,
+            self.metaindex[Columns.LONGITUDE.value].values,
+            Coordinates(np.array([latitude]), np.array([longitude])),
+            rank,
+        )
+
+        self.metaindex = self.metaindex.iloc[indices_nearest_neighbours.ravel(), :]
+
+        required_station_groups = [
+            DwdObservationRoadWeatherStationGroups(grp)
+            for grp in self.metaindex.loc[
+                :, Columns.STATION_GROUP.value
+            ].drop_duplicates()
+        ]
+
+        _dat = [
+            self._collect_data_by_station_group(road_weather_station_group)
+            for road_weather_station_group in required_station_groups
+        ]
+        road_weather_station_data = pd.concat(
+            _dat,
+        )
+
+        return road_weather_station_data[
+            road_weather_station_data.index.get_level_values("shortStationName").isin(
+                self.metaindex[Columns.STATION_ID.value].tolist()
+            )
+        ]
+
+    def _collect_data_by_station_name(self, station_name: str) -> pd.DataFrame:
+        """ Takes station_name to download and parse RoadWeather Station data """
+        self.metaindex = self.metaindex.loc[
+            self.metaindex[Columns.NAME.value] == station_name, :
+        ]
+
+        required_station_groups = [
+            DwdObservationRoadWeatherStationGroups(grp)
+            for grp in self.metaindex.loc[
```

Comment:
for grp in self.metaindex[Columns.STATION_GROUP.value].unique()

be aware of the index when using .drop_duplicates()